### PR TITLE
rune/libenclave/attestation: move the attestation sub-packages internal

### PR DESCRIPTION
**Pull request by hustliyilin**
1. rune/libenclave/intelsgx: redefine quote-length-related constants for both epid and ecdsa.
2. rune/libenclave/intelsgx && attestation: move the attestation/sgx/attestation.go to the intelsgx package
3. rune/libenclave/attestation: move sgx and registration packages into the internal.

Fixes: alibaba#613
Signed-off-by: Yilin Li <YiLin.Li@linux.alibaba.com>

### CODE REVIEW

File: rune/libenclave/attestation/sgx/attester/attester-ecdsa.go [hunk 1]
```diff
@@ -3,15 +3,14 @@ package sgx_attester // import "github.com/inclavare-containers/rune/libenclave/
 import (
 	"fmt"
 	"github.com/inclavare-containers/rune/libenclave/attestation/registration"
-	"github.com/inclavare-containers/rune/libenclave/attestation/sgx"
```
Comment (jiazhang0) [inline rune/libenclave/attestation/sgx/attester/attester-ecdsa.go:6]: Why not keep it in attestation package?

File: rune/libenclave/intelsgx/pccs/pccs.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
+package pccs // import "github.com/inclavare-containers/rune/libenclave/intelsgx/pccs"
+
+/*
+#cgo LDFLAGS: -lsgx_dcap_quoteverify
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include "sgx_dcap_quoteverify.h"
```
Comment (jiazhang0) [inline rune/libenclave/intelsgx/pccs/pccs.go:8]: Where is this header file? Is it placed under the standard path?

File: rune/libenclave/attestation/sgx/attester/attester-ecdsa.go (hunk 1 above)
Comment (hustliyilin) [reply rune/libenclave/attestation/sgx/attester/attester-ecdsa.go:6]: > Why not keep it in attestation package?

If we want to keep `attestation` sub-packages internal, we must keep the `attestation/sgx` unimportable. so we need to move constants of the `attestation/sgx` package into the `intelsgx` package

File: rune/libenclave/intelsgx/pccs/pccs.go (hunk 2 above)
Comment (hustliyilin) [reply rune/libenclave/intelsgx/pccs/pccs.go:8]: > Where is this header file? Is it placed under the standard path?

Yes, We introduce the new dependence(`libsgx-dcap-quote-verify-*.x86_64` rpm package) for `sgx-tools` and `rune` in the line 4.

Comment (hustliyilin) [reply rune/libenclave/intelsgx/pccs/pccs.go:8]: I'll add the new dependence(`libsgx-dcap-quote-verify-*.x86_64`)  for the spec , deb, README.md and CI/CD for both `rune` and `sgx-tools` later.

File: rune/libenclave/attestation/sgx/attester/attester-ecdsa.go (hunk 1 above)
Comment (jiazhang0) [reply rune/libenclave/attestation/sgx/attester/attester-ecdsa.go:6]: Is there any other packages requiring ias package? If not, move it to internal as well.

File: rune/libenclave/intelsgx/pccs/pccs.go (hunk 2 above)
Comment (jiazhang0) [reply rune/libenclave/intelsgx/pccs/pccs.go:8]: You are using "" to include the header file, so it looks like sgx_dcap_quoteverify.h is not installed by libsgx-dcap-quote-verify-*.x86_64 to a standard include file location.

Comment (hustliyilin) [reply rune/libenclave/intelsgx/pccs/pccs.go:8]: > You are using "" to include the header file, so it looks like sgx_dcap_quoteverify.h is not installed by libsgx-dcap-quote-verify-*.x86_64 to a standard include file location.

Ok, I'll change it to `<>`.

File: rune/libenclave/attestation/sgx/attester/attester-ecdsa.go (hunk 1 above)
Comment (hustliyilin) [reply rune/libenclave/attestation/sgx/attester/attester-ecdsa.go:6]: > Is there any other packages requiring ias package? If not, move it to internal as well.

OK, but I also need to move PCCS internal.

**Comment by hustliyilin** [comment]
`Pccs` and `challenger-ecdsa` will be placed another pr,  This pr is just for making `attestation` sub-packages internal. @jiazhang0

**Pull request merged**